test failure: stack overflow calling "gcd"

Relevant portion of the test output:

```
--- FAIL: TestAll (0.39s)
...
    ivy_test.go:47: function.ivy
    ivy_test.go:95:
        execution failure ( :11: stack overflow calling "gcd"
        ) at testdata/function.ivy:199:
        op fac n =
         n <= 1 : 1
         n * fac n - 1

        op a gcd b =
         a == b: a
         a > b: b gcd a-b
         a gcd b-a

        fac 10
        1562 gcd fac 11
FAIL
FAIL    robpike.io/ivy  0.615s
FAIL
```

Git bisect says that commit f024fcaf177 introduced the failure, and tests pass with it reverted. I can't see why.

This is with Go 1.21.0 on an Ubuntu Linux VM. I can't reproduce the failure on macOS, OpenBSD, or Alpine Linux.

--- a/ivy_test.go
+++ b/ivy_test.go
@@ -180,6 +180,7 @@ func reset() {
 	testConf.SetFormat("")
 	testConf.SetMaxBits(1e9)
 	testConf.SetMaxDigits(1e4)
+	testConf.SetMaxStack(100000)
 	testConf.SetOrigin(1)
 	testConf.SetPrompt("")
 	testConf.SetBase(0, 0)

